### PR TITLE
Better naming for controllers deployment

### PR DESCRIPTION
**Pull request by enxebre**
Better naming for controllers deployment

### CODE REVIEW

File: owned-manifests/machine-api-controllers.yaml [hunk 1]
```diff
@@ -36,18 +36,18 @@ spec:
         key: node.alpha.kubernetes.io/unreachable
         operator: Exists
       containers:
-      - name: controller-manager
+      - name: machineset-controller-manager
         image: {{ .Controllers.Provider }}
         command:
-          - "./manager"
+          - /manager
```
Comment (michaelgugino) [inline owned-manifests/machine-api-controllers.yaml]: Was this intentional?

Comment (enxebre) [reply owned-manifests/machine-api-controllers.yaml]: yes, to align with all other `command` and other fields syntax

File: owned-manifests/machine-api-controllers.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: clusterapi-manager-controllers
+  name: machine-api-controllers
```
Comment (frobware) [inline owned-manifests/machine-api-controllers.yaml]: +1 \o/

File: owned-manifests/machine-api-controllers.yaml [hunk 3]
```diff
@@ -36,18 +36,18 @@ spec:
         key: node.alpha.kubernetes.io/unreachable
         operator: Exists
       containers:
-      - name: controller-manager
+      - name: machineset-controller-manager
```
Comment (frobware) [inline owned-manifests/machine-api-controllers.yaml]: Huzzah. Though I still question the `-manager` bit. Kind of vacuous really. If it's a controller, what does the manager bit mean?

Comment (enxebre) [reply owned-manifests/machine-api-controllers.yaml]: These are actually controller managers that can hold 1..N controllers

Comment (frobware) [reply owned-manifests/machine-api-controllers.yaml]: > These are actually controller managers that can hold 1..N controllers

Given what you just wrote shouldn't this remain as 'controller-manager'.

File: vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/operators/machine-api-operator.go [hunk 4]
```diff
@@ -23,7 +23,7 @@ var _ = g.Describe("[Feature:Operators] Machine API operator deployment should",
 		client, err := e2e.LoadClient()
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		deploymentName := "clusterapi-manager-controllers"
+		deploymentName := "machine-api-controllers"
```
Comment (bison) [inline vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/operators/machine-api-operator.go]: Is there a PR for this on cluster-api-actuator-pkg?

**Comment by enxebre** [comment]
we are hitting https://bugzilla.redhat.com/show_bug.cgi?id=1698624 re-triggering for CI measurement

**Comment by ingvagabund** [comment]
This PR is no longer relevant. We have already a PR that moves the yaml manifest under the code: 
https://github.com/openshift/machine-api-operator/pull/280

File: owned-manifests/machine-api-controllers.yaml (hunk 3 above)
Comment (ingvagabund) [reply owned-manifests/machine-api-controllers.yaml]: `machineset`? That's very confusing. The `manager` binary runs more than the machineset controller. It also runs the machinedeployment controller and node controller. It does not make sense to reduce just to machineset.

File: owned-manifests/machine-api-controllers.yaml [hunk 5]
```diff
@@ -36,18 +36,18 @@ spec:
...
+          - /manager
         args:
           - --logtostderr=true
           - --v=3
         resources:
           requests:
             cpu: 10m
             memory: 20Mi
-      - name: machine-controller
+      - name: machine-controller-manager
```
Comment (ingvagabund) [inline owned-manifests/machine-api-controllers.yaml]: -1. Either `controller` or just `manager`. Yes, we built on top of controller-runtime which defines a concept of a manager. Yet, we run only one controller. It only make sense to either use `machine-controller` or `machine-manager`. What's the motivation behind the renaming? `Better naming` does not explain it much.

File: owned-manifests/machine-api-controllers.yaml [hunk 6]
```diff
@@ -70,7 +70,7 @@ spec:
           requests:
             cpu: 10m
             memory: 20Mi
-      - name: machine-healthcheck
+      - name: machine-healthcheck-controller-manager
```
Comment (ingvagabund) [inline owned-manifests/machine-api-controllers.yaml]: The same here. Either `machine-healthcheck-controller` or `machine-healthcheck-manager`. Not both.

**Comment by ingvagabund** [comment]
Unless, there is an issue fixed by this PR, let's wait for https://github.com/openshift/machine-api-operator/pull/280 to merge and then rebase.

File: vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/operators/machine-api-operator.go (hunk 4 above)
Comment (enxebre) [reply vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/operators/machine-api-operator.go]: https://github.com/openshift/cluster-api-actuator-pkg/pull/63

File: owned-manifests/machine-api-controllers.yaml (hunk 5 above)
Comment (enxebre) [reply owned-manifests/machine-api-controllers.yaml]: *-controller-manager is the most accurate, self explanatory and in line with core components naming https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/

Comment (ingvagabund) [reply owned-manifests/machine-api-controllers.yaml]: You refer to only one component name that ends with `-controller-manager` suffix. How does it generalizes to other component names? Plus, name `kube-controller-manager` is very pre-historic name from the middle ages of kubernetes when we used swords and sticks to do all the magic.

Comment (enxebre) [reply owned-manifests/machine-api-controllers.yaml]: not sure I follow, this is literally a controller manager that might wrap one or more Controllers

Comment (enxebre) [reply owned-manifests/machine-api-controllers.yaml]: e.g https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/cmd/clusterctl/examples/aws/provider-components-base/kustomization.yaml#L10
e.g https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/

**Comment by enxebre** [comment]
this should go green now https://github.com/openshift/machine-api-operator/pull/286
/retest

**Comment by enxebre** [comment]
```Installing from release registry.svc.ci.openshift.org/ci-op-mnr7l5cy/release@sha256:8e82a0c9df80984abe9b72587a95a8ca32a6a5ac8ff6d3cc892369a792a014fa
level=warning msg="Found override for ReleaseImage. Please be warned, this is not advised"
level=info msg="Consuming \"Install Config\" from target directory"
level=info msg="Creating infrastructure resources..."
level=error
level=error msg="Error: Error applying plan:"
level=error
level=error msg="1 error occurred:"
level=error msg="\t* module.vpc.aws_route.igw_route: 1 error occurred:"
level=error msg="\t* aws_route.igw_route: Error creating route: timeout while waiting for state to become 'success' (timeout: 2m0s)"
level=error
level=error
level=error
level=error
```
/retest

File: owned-manifests/machine-api-controllers.yaml (hunk 5 above)
Comment (enxebre) [reply owned-manifests/machine-api-controllers.yaml]: changed to *-controller discussed offline to be more clear for the majority

**Pull request merged**